index.js

```diff
@@ -62,8 +62,15 @@ function lint(webpackInstance, input, options) {
 
   var program;
   if (options.typeCheck) {
-    var tsconfigPath = resolveFile(options.tsConfigFile);
-    program = Lint.Linter.createProgram(tsconfigPath);
+    if (!webpackInstance.options.tslintProgram) {
```

Comment:
`compiler.options` [is deprecated](https://webpack.js.org/api/loaders/#this-options).